### PR TITLE
Remove invalid Markdown backticks in std.path

### PR DESCRIPTION
**Pull request by wilzbach**
To fix https://github.com/dlang/dmd/pull/8325

### CODE REVIEW

**Comment by wilzbach** [comment]
I'm not sure I understand, here's the diff from DAutoTest of the resulting HTML:

```diff
-    Note that on Windows, both the backslash (<span class="d_inlinecode donthyphenate notranslate"></span>\<span class="d_inlinecode donthyphenate notranslate"></span>) and the slash (<span class="d_inlinecode donthyphenate notranslate"></span>/<span class="d_inlinecode donthyphenate notranslate"></span>)
+    Note that on Windows, both the backslash (<span class="d_inlinecode donthyphenate notranslate">\</span>) and the slash (<span class="d_inlinecode donthyphenate notranslate">/</span>)
     are in principle valid directory separators.  This module treats them
     both on equal footing, but in cases where a <i>new</i> separator is
     added, a backslash will be used.  Furthermore, the <a href="#buildNormalizedPath"><span class="d_inlinecode donthyphenate notranslate">buildNormalizedPath</span></a>
```

**Comment by wilzbach** [comment]
BTW with a quick grep I found more instances of double-backticks.

**Comment by jmdavis** [comment]
> I'm not sure I understand, here's the diff from DAutoTest of the resulting HTML:

Are you referring to my comment about using

    $(D `\`)

instead? I mean that both slashes are supposed to be strings. Originally, they would have been

    $(D `/`)

and

    $(D `\`)

where the backticks make them strings where the backslash doesn't have to be escaped - e.g. `$(D "\\")`. However, someone came along and mass-replaced `$(D ... )` with backslashes, which then screwed up any case where backslahes were used as string delimeters. The way to fix this is not to get rid of the "extra" backticks but to stop using backticks instead of `$(D ...)`. If ddoc can't handle backtick strings in `$(D ...)` anymore, then that's a serious bug in ddoc that needs to be fixed.

> BTW with a quick grep I found more instances of double-backticks.

Any case where backticks were used as string delimiters would have been screwed up by any attempt to mass-replace `$(D ...)` with backticks, so it doesn't surprise me at all if we ended up with double backticks in a number of places.

**Comment by dgileadi** [comment]
As the DAutoTest output above shows the consecutive backticks are not currently producing the desired output—they surround the slash with two empty code spans instead of wrapping it in a code span.

Markdown [collapses consecutive backticks](https://daringfireball.net/projects/markdown/syntax#code), so my upcoming markdown support will treat \`\`\\\`\` and \`\\\` identically and would thus produce the correct output. The problem in dlang/dmd#8325 would be fixed by the upcoming Markdown support for collapsing consecutive backticks. However the `markdown` branch doesn't yet support consecutive backticks but does support backslash escapes, and that caused the failure. It seems to me that this PR is the correct fix for dlang/dmd#8325, and also has the happy effect of producing the desired output in `master`.

**Comment by jmdavis** [comment]
> Markdown collapses consecutive backticks, so my upcoming markdown support will treat ``\`` and `\` identically and would thus produce the correct output.

When you say that, do you mean that two backticks in a row are treated the same as one backtick or that two backticks in a row will result in the output having a single backtick? What we want here is to treat these as strings - the equivalent of `"/"` and `"\\"` except that it uses backticks for string delimiters instead of double quotes so that the escaping isn't necessary. If there's a clean way to do that with escaping the backticks rather than using `$(D...)`, then fine, but these should be string literals with backticks in the final result.

Either way, whatever you do with markdown support, `$(D...)` with backticks in it still needs to treat them as backticks and not markdown, or it will break all of the existing inline code samples that use backticks for string literals.

**Comment by dgileadi** [comment]
>> Markdown collapses consecutive backticks, so my upcoming markdown support will treat \'\'\\'\' and \'\\' identically and would thus produce the correct output.

> When you say that, do you mean that two backticks in a row are treated the same as one backtick or that two backticks in a row will result in the output having a single backtick? What we want here is to treat these as strings - the equivalent of "/" and "\\" except that it uses backticks for string delimiters instead of double quotes so that the escaping isn't necessary. If there's a clean way to do that with escaping the backticks rather than using $(D...), then fine, but these should be string literals with backticks in the final result.

Yes, under Markdown multiple backticks in a row will be treated the same as a single backtick for creating a code span, provided the span is closed with the same number of backticks. If you want to output a backtick character you can backslash-escape it like \\`.

Note that for the documentation this PR addresses in std.path it appears that the backticks are not currently displayed (see the red line in the DAutoTest output), and from the context of the paragraph I don't believe that the original author wanted a backtick character to display. Rather I believe they intended to make code spans that displayed a backslash and a forward slash.

> Either way, whatever you do with markdown support, $(D...) with backticks in it still needs to treat them as backticks and not markdown, or it will break all of the existing inline code samples that use backticks for string literals.

So far as I can tell, $(D...) is not treated specially in ddoc processing—backticks inside it are treated just the same as backticks outside it. So today $(D \`stuff in backticks\`) would be double-wrapped by two code spans.

**Comment by aG0aep6G** [comment]
> I don't believe that the original author wanted a backtick character to display. Rather I believe they intended to make code spans that displayed a backslash and a forward slash.

Nope. The backticks were supposed to be displayed. @CyberShadow changed it from

    '\'

to

    `\`

in #1267.

The PR description says: "Fixed a few char/string literals so they're in valid D syntax". A backslash on its own isn't valid D syntax. But a backslash in backticks is a string literal.

That was back in 2.063. The backticks were displayed then. DDoc only got backtick support with 2.067 (https://github.com/dlang/dmd/commit/26c5de42214c788ab42d96694a994c3eb8ae976a).

**Comment by dgileadi** [comment]
Nice research. In that case maybe it has been broken since 2.067?

**Comment by aG0aep6G** [comment]
> In that case maybe it has been broken since 2.067?

Yearp.

**Comment by wilzbach** [comment]
So what's the consensus here: do we go with `` `/` `` or `` $(D `/`) ``?
I don't really mind and it seems to me to be a purely aesthetic choice. I just want one solution to get merged, s.t. the DMD PR is no longer blocked.

**Comment by CyberShadow** [comment]
> I don't really mind and it seems to me to be a purely aesthetic choice.

It's not. The intended result is that in the rendered documentation, a string literal is displayed. The contents of the backticks is not D source code, so it should not be highlighted as such. You broke it in #6391, so, please take the opportunity to fix it.

This should have been caught by DAutoTest's HTML diff before #6391 was merged, and to this day I still don't understand why it didn't work there, or why #6391 was merged before ensuring that the changes were correct.

**Comment by wilzbach** [comment]
Changed it to `` $(D `/`) `` then.

File: std/path.d [hunk 1]
```diff
@@ -1425,7 +1425,7 @@ private auto _withDefaultExtension(R, C)(R path, C[] ext)
     preceding segments will be dropped.
 
     On Windows, if one of the path segments are rooted, but not absolute
-    (e.g. ``\foo``), all preceding path segments down to the previous
+    (e.g. `\foo`), all preceding path segments down to the previous
```
Comment (jmdavis) [inline std/path.d]: $(D `\foo`)

**Comment by RazvanN7** [comment]
ping @wilzbach

**Comment by wilzbach** [comment]
Finally got around addressing the final nits. We should be good to go here now?

**Pull request merged**